pkg/controllers/binding/common.go

```diff
@@ -96,13 +108,27 @@ func ensureWork(c client.Client, workload *unstructured.Unstructured, overrideMa
 		workLabel := mergeLabel(clonedWorkload, workNamespace, binding, scope)
 
 		if clonedWorkload.GetKind() == util.DeploymentKind && (referenceRSP != nil || hasScheduledReplica) {
-			err = applyReplicaSchedulingPolicy(clonedWorkload, desireReplicaInfos[targetCluster.Name])
+			err = applyReplicaSchedulingPolicy(clonedWorkload, desireReplicaInfos[targetCluster.Name], util.ReplicasField)
 			if err != nil {
 				klog.Errorf("failed to apply ReplicaSchedulingPolicy for %s/%s/%s in cluster %s, err is: %v",
 					clonedWorkload.GetKind(), clonedWorkload.GetNamespace(), clonedWorkload.GetName(), targetCluster.Name, err)
 				return err
 			}
 		}
+		if clonedWorkload.GetKind() == util.JobKind && hasScheduledReplica {
+			err = applyReplicaSchedulingPolicy(clonedWorkload, desireReplicaInfos[targetCluster.Name], util.ParallelismField)
+			if err != nil {
+				klog.Errorf("failed to apply Parallelism for %s/%s/%s in cluster %s, err is: %v",
+					clonedWorkload.GetKind(), clonedWorkload.GetNamespace(), clonedWorkload.GetName(), targetCluster.Name, err)
+				return err
+			}
+			err = applyReplicaSchedulingPolicy(clonedWorkload, int64(jobCompletions[i].Replicas), util.CompletionsField)
+			if err != nil {
+				klog.Errorf("failed to apply Completions for %s/%s/%s in cluster %s, err is: %v",
+					clonedWorkload.GetKind(), clonedWorkload.GetNamespace(), clonedWorkload.GetName(), targetCluster.Name, err)
+				return err
+			}
+		}
```

Comment:
I smell some trouble if we want to include `Job` in #923, because usually, a workload has one `replica` filed, but Job has two...

Reply:
Emmm, it's awkward. 😂 Actually job only has one replica field `.spec.parallelism`, but the `.spec.completions` is related to this field... I have no idea, maybe the job should be treated independently.